include/integratorxx/quadratures/gausslegendre.hpp

```diff
@@ -18,18 +18,44 @@ class GaussLegendre :
   using weight_type      = typename base_type::weight_type;
   using point_container  = typename base_type::point_container;
   using weight_container = typename base_type::weight_container;
-  
-  GaussLegendre(size_t npts, point_type lo, point_type up):
-    base_type( npts, lo, up ) { }
+
+  GaussLegendre(size_t npts):
+    base_type( npts ) { }
 
   GaussLegendre( const GaussLegendre& ) = default;
   GaussLegendre( GaussLegendre&& ) noexcept = default;
 };
 
-
-
-
-
+  /**
+   *  Evaluates the Legendre polynomial at z with the recurrence
+   *  relation \f$(n+1) P_{n+1}(x) = (2n+1) x P_{n}(x) - n P_{n-1}(x)
+   *  \f$, and its derivative with the relation \f$ \frac {x^2-1} {n}
+   *  \frac {\rm d} {{\rm d}x} P_{n}(x) = x P_{n}(x) - P_{n-1}(x) \f$.
+   *
+   *  Inputs:
+   *  x - x coordinate
+   *  n - order of polynomial
+   *  p_n - value of the Legendre polynomial, P_{n}(x)
+   *  dp_n - value of the derivative of the Legendre polynomial, dP_{n}/dx
+   */
+  template <typename PointType> inline void eval_Pn(PointType x, size_t n, PointType & p_n, PointType & dp_n) {
```

Comment:
Implicit return by parameter reference is an anti pattern in modern C++, please return these values either as a pair or a tuple.

Reply:
I am still hoping to learn more modern C++. Switched to `std::pair`.